src/methods.cpp

```diff
@@ -28,15 +30,26 @@ extern "C"
     static PyObject* region_begin(PyObject* self, PyObject* args)
     {
         const char* module;
-        const char* region_name;
+        const char* function_name;
         const char* file_name;
+        PyObject* identifier = nullptr;
         std::uint64_t line_number = 0;
 
-        if (!PyArg_ParseTuple(args, "sssK", &module, &region_name, &file_name, &line_number))
+        if (!PyArg_ParseTuple(args, "sssK|O", &module, &function_name, &file_name, &line_number,
```

Comment:
Judging from the use the identifier argument is not optional, but it can be None, which is different. I'd make it non-optional which also simplifies the below check

Reply:
d

Reply:
d